lib/reverse_sentence.rb

```diff
@@ -1,6 +1,30 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) - the n is because the number of times the loop runs is dependant on the length of my_sentence.
+# Space complexity: O(n) - array created by split method. After reversing the words, join method used to make it into a single string again.
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+
+  if my_sentence.nil?
+    return nil
+  else
+    word_array = my_sentence.split(/\s/) # regex not working for test #6
+    i = 0
+    last = -1
+    (word_array.length/2).times do |i|
+      temp = word_array[i]
+      word_array[i] = word_array[last]
+      word_array[last] = temp
+      last -= 1
+    end
+  end
+  new_string = word_array.join(" ")
+
+  my_sentence.length.times do |i|
+    my_sentence[i] = new_string[i]
```

Comment:
You are crashing here as `my_sentence[i]` is `nil` and you are trying to assign it to a string, trying to assign something past the end of the string to a string value.